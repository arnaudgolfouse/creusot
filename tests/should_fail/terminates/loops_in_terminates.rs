#![allow(unused)]
extern crate creusot_contracts;
use creusot_contracts::*;

#[check(terminates)]
fn terminates_while_loop() {
    #[allow(while_true)]
    while true {}
}

#[check(terminates)]
fn terminates_loop_loop() {
    loop {}
}
