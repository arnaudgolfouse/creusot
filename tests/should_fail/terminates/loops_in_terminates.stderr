error: `#[check(terminates)]` function must not contain loops without variants.
 --> loops_in_terminates.rs:6:1
  |
6 | fn terminates_while_loop() {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
7 |     #[allow(while_true)]
8 |     while true {}
  |     - help: Add a variant here: `#[variant(...)]`
  |
note: looping occurs here
 --> loops_in_terminates.rs:8:5
  |
8 |     while true {}
  |     ^^^^^^^^^^^^^

error: `#[check(terminates)]` function must not contain loops without variants.
  --> loops_in_terminates.rs:12:1
   |
12 | fn terminates_loop_loop() {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
13 |     loop {}
   |     - help: Add a variant here: `#[variant(...)]`
   |
note: looping occurs here
  --> loops_in_terminates.rs:13:5
   |
13 |     loop {}
   |     ^^^^^^^

error: aborting due to 2 previous errors

